From ed2579fbab8c8fd860acf448cef33e0024b56f6b Mon Sep 17 00:00:00 2001
From: Andrei Karas <akaras@inbox.ru>
Date: Mon, 29 Nov 2010 23:34:16 +0200
Subject: [PATCH 060/226] Add client ip addresse in logs.

As in tmw server commit cc3f6547545edc7a5cbf2d82c4c2bcb405582bd5.
---
 src/char/char.c | 26 ++++++++++++++++++++------
 1 file changed, 20 insertions(+), 6 deletions(-)

diff --git a/src/char/char.c b/src/char/char.c
index 787d787..1568e88 100644
--- a/src/char/char.c
+++ b/src/char/char.c
@@ -1078,12 +1078,19 @@ int make_new_char (int fd, unsigned char *dat)
             online_chars[j] = -1;
     }
 
+    char ip[17];
+    unsigned char *sin_addr =
+        (unsigned char *) &session[fd]->client_addr.sin_addr;
+    ip[16] = 0;
+    sprintf (ip, "%d.%d.%d.%d", sin_addr[0], sin_addr[1], sin_addr[2],
+             sin_addr[3]);
+
     char_log
-        ("Creation of New Character: (connection #%d, account: %d) slot %d, character Name: %s, stats: %d+%d+%d+%d+%d+%d=%d, hair: %d, hair color: %d."
+        ("Creation of New Character: (connection #%d, account: %d) slot %d, character Name: %s, stats: %d+%d+%d+%d+%d+%d=%d, hair: %d, hair color: %d. [%s]"
          RETCODE, fd, sd->account_id, dat[30], dat, dat[24], dat[25], dat[26],
          dat[27], dat[28], dat[29],
          dat[24] + dat[25] + dat[26] + dat[27] + dat[28] + dat[29], dat[33],
-         dat[31]);
+         dat[31], ip);
 
     memset (&char_dat[i], 0, sizeof (struct mmo_charstatus));
 
@@ -3209,6 +3216,13 @@ int parse_char (int fd)
                 if (!sd || RFIFOREST (fd) < 3)
                     return 0;
 
+                char ip[17];
+                ip[16] = 0;
+                unsigned char *sin_addr =
+                    (unsigned char *) &session[fd]->client_addr.sin_addr;
+                sprintf (ip, "%d.%d.%d.%d", sin_addr[0], sin_addr[1],
+                         sin_addr[2], sin_addr[3]);
+
                 // if we activated email creation and email is default email
                 if (email_creation != 0 && strncmp (sd->email, "a@a.com", 40) == 0
                     && login_fd > 0)
@@ -3229,9 +3243,9 @@ int parse_char (int fd)
                     if (ch != 9)
                     {
                         char_log
-                            ("Character Selected, Account ID: %d, Character Slot: %d, Character Name: %s."
+                            ("Character Selected, Account ID: %d, Character Slot: %d, Character Name: %s [%s]"
                              RETCODE, sd->account_id, RFIFOB (fd, 2),
-                             char_dat[sd->found_char[ch]].name);
+                             char_dat[sd->found_char[ch]].name, ip);
                         // searching map server
                         i = search_mapserver (char_dat
                                               [sd->found_char[ch]].last_point.
@@ -3335,9 +3349,9 @@ int parse_char (int fd)
                                 char_dat[sd->found_char[ch]].last_point.map,
                                 16);
                         printf
-                            ("Character selection '%s' (account: %d, slot: %d).\n",
+                            ("Character selection '%s' (account: %d, slot: %d) [%s]\n",
                              char_dat[sd->found_char[ch]].name,
-                             sd->account_id, ch);
+                             sd->account_id, ch, ip);
                         printf ("--Send IP of map-server. ");
                         if (lan_ip_check (p))
                             WFIFOL (fd, 22) = inet_addr (lan_map_ip);
-- 
2.1.0

